File: src/CSharp/283.MoveZeroes.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.MoveZeroes
{
public class Solution {
    public void MoveZeroes(int[] nums) {
        int j = 0;
        int n = nums.Length;
        for(int i = 0; i < n; i++)
        {
            if (nums[i] != 0){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                ++j;
            }
        }
    }
}
}

File: src/Java/283.MoveZeroes.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

class Solution {
    public void moveZeroes(int[] nums) {
        int j = 0;
        int n = nums.length;
        for(int i = 0; i < n; i++)
        {
            if (nums[i] != 0){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                ++j;
            }
        }
    }
}
